cvat-core/src/cloud-storage.js

```
@@ -0,0 +1,407 @@
+                     * @instance
+                     * @throws {module:API.cvat.exceptions.ArgumentError}
+                     */
+                    token: {
+                        get: () => data.token,
+                        set: (value) => {
+                            if (typeof value === 'string') {
+                                if (value.trim().length) {
+                                    data.token = value;
+                                } else {
+                                    throw new ArgumentError('Value must not be empty');
+                                }
+                            }
+                        },
+                    },
+                    /**
+                     * Unique resource name
+                     * @name resourceName
+                     * @type {string}
+                     * @memberof module:API.cvat.classes.CloudStorage
+                     * @instance
+                     * @throws {module:API.cvat.exceptions.ArgumentError}
+                     */
+                    resourceName: {
+                        get: () => data.resourceName,
+                        set: (value) => {
+                            if (!value.trim().length) {
+                                throw new ArgumentError('Value must not be empty');
+                            }
+                            data.resourceName = value;
+                        },
+                    },
+                    /**
+                     * @name manifestPath
+                     * @type {string}
+                     * @memberof module:API.cvat.classes.CloudStorage
+                     * @instance
+                     */
+                    manifestPath: {
+                        get: () => data.manifestPath,
+                        set: (value) => {
+                            if (typeof value !== undefined) {
+                                data.manifestPath = value;
+                            }
```

If undefined is a wrong value, I think we should raise an exception